### PR TITLE
Added feature to auto hyphenate text at syllable boundaries

### PR DESCRIPTION
**Pull request by jrief**
You can hyphenate the text entered into the editor, so that the HTML entity ``&shy;`` [soft-hyphen](https://github.com/datadesk/django-softhyphen) automatically is added in between words, at the correct syllable boundary.
This is quite handy, specially when using narrow columns.

### CODE REVIEW

**Comment by yakky** [comment]
This is super awesome!

**Comment by yakky** [comment]
I had to revert this as triggers errors in the django CMS test suite. Could you please double check?

**Comment by jrief** [comment]
@yakky has been fixed.

Some other consideration: Should we allow soft-hyphening per instance? This would require an additional Boolean field on the model and a checkbox nearby the text editor to activate this.

**Comment by yakky** [comment]
@jrief thanks!
I think that the per-instance flag can be left to extended plugins as it would be too implementation dependent.
Probably we can add an example to implement this feature in external plugins via AbstractText

File: djangocms_text_ckeditor/models.py [hunk 1]
```diff
@@ -40,6 +45,8 @@ def save(self, *args, **kwargs):
         body = self.body
         body = extract_images(body, self)
         body = clean_html(body, full=False)
+        if settings.TEXT_AUTO_HYPHENATE:
+            body = hyphenate(body, language=self.cmsplugin_ptr.language)
```
Comment (yakky) [inline djangocms_text_ckeditor/models.py]: why using ``cms.cmsplugin_ptr.language``? Isn't ``self.language`` available?

**Pull request merged**